samples/iree-reduce/softmax.mlir

```diff
@@ -0,0 +1,91 @@
+// RUN: iree-compile %s --iree-hal-target-backends=cuda \
+// RUN:     --iree-opt-const-expr-hoisting=false --iree-opt-const-eval=false \
+// RUN:     --iree-codegen-llvmgpu-enable-transform-dialect-jit=false \
+// RUN:     --iree-flow-dispatch-use-transform-dialect=%p/softmax_dispatch_spec.mlir \
+// RUN:     --iree-codegen-llvmgpu-use-transform-dialect=%p/softmax_codegen_spec.mlir | \
+// RUN: iree-run-module --module=- --function=softmax --device=cuda | \
```

Comment:
We shouldn't have basic e2e tests that rely on a gpu. Can this be reformulated to work on CPU? Or maybe just added as a example.